backend/services/implementations/courseService.ts

```diff
@@ -0,0 +1,121 @@
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+import {
+  CourseRequestDTO,
+  CourseResponseDTO,
+  ICourseService,
+} from "../interfaces/ICourseService";
+import MgCourse, { Course } from "../../models/course.model";
+
+const Logger = logger(__filename);
+
+class CourseService implements ICourseService {
+  async getCourse(id: string): Promise<CourseResponseDTO> {
+    let course: Course | null;
+    try {
+      course = await MgCourse.findById(id);
+      if (!course) {
+        throw new Error(`Course id ${id} not found`);
+      }
+    } catch (error: unknown) {
+      Logger.error(`Failed to get course. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return {
+      id: course.id,
+      title: course.title,
+      description: course.description,
+      image: course.image,
+      previewImage: course.previewImage,
+      lessons: course.lessons,
+    };
+  }
+
+  async getCourses(): Promise<CourseResponseDTO[]> {
+    try {
+      const courses: Array<Course> = await MgCourse.find();
+      return courses.map((course) => ({
+        id: course.id,
+        title: course.title,
+        description: course.description,
+        image: course.image,
+        previewImage: course.previewImage,
+        lessons: course.lessons,
+      }));
+    } catch (error: unknown) {
+      Logger.error(`Failed to get courses. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+  }
+
+  async createCourse(course: CourseRequestDTO): Promise<CourseResponseDTO> {
```

Comment:
Check if `create` operation returns null if it fails (aka `Promise<CourseRequestDTO>` vs. `Promise<CourseRequestDTO | null>`)

Reply:
Yeah I'm not a big fan of how starter-code implements it. Like, `newCourse` will always be a Course type, never null. If it is null, it's because an error was thrown and it will never be returned anyway. My vote would be to return null if it fails

Reply:
As in, `newCourse` will never be null even when it fails because it'll throw an error?